Tests/Tests/IndexManagerTests.m

```diff
@@ -839,6 +843,117 @@ - (void)testNewIndexManagerPicksUpDocumentsAddedBeforeEnsureIndexedDuringQuery
     [[NSFileManager defaultManager] removeItemAtPath:factoryPath error:nil];
 }
 
+
+/**
+ Set up following document structure
+ 
+                winner (returned by getDocumentWithId:...)
+                |
+        /---- 2-4f
+ 1-97 --
+        \---- 2-33 ---- 3-bc
+                         |
+                         deleted (returned by _changes)
+ 
+ This test ensures we index the winning revision rather than the one with the highest 
+ revision id.
+
+ The bug we are fixing (BugId: 45667) results from the indexing code updating the
+ index used the revision the _changes feed provides. It's possible the changed
+ feed gives a revision which isn't the winning revision. For example, it gives
+ 3-bc for the tree above, rather than 204f.
+ */
+- (void)test_CanIndexConflictedDocWithLatestRevDeleted
+{
+    CDTDatastore *datastore;
+    
+    NSString *factoryPath = [self createTemporaryDirectoryAndReturnPath];
```

Comment:
Does this test need to do any cleanup (because it creates its own factory) or is this handled in teardown?